src/main/kotlin/org/rust/ide/hints/RustParameterInfoHandler.kt

```diff
@@ -0,0 +1,159 @@
+
+    override fun updateParameterInfo(place: PsiElement, context: UpdateParameterInfoContext) {
+        val argIndex = findArgumentIndex(place)
+        if (argIndex == INVALID_INDEX) {
+            context.removeHint()
+            return
+        }
+        context.setCurrentParameter(argIndex)
+        if (context.parameterOwner == null) {
+            context.parameterOwner = place
+        } else if (context.parameterOwner != findElementForParameterInfo(place)) {
+            context.removeHint()
+            return
+        }
+        context.objectsToView.mapIndexed { i, o -> context.setUIComponentEnabled(i, true) }
+    }
+
+    override fun updateUI(p: RustArgumentsDescription?, context: ParameterInfoUIContext) {
+        if (p == null) {
+            context.isUIComponentEnabled = false
+            return
+        }
+        val range = p.getArgumentRange(context.currentParameterIndex)
+        hintText = p.presentText
+        context.setupUIComponentPresentation(
+            hintText,
+            range.startOffset,
+            range.endOffset,
+            !context.isUIComponentEnabled,
+            false,
+            false,
+            context.defaultParameterColor)
+    }
+
+    /**
+     * Finds index of the argument in the given place
+     */
+    private fun findArgumentIndex(place: PsiElement): Int {
+        val callArgs = place.parentOfType<RustArgListElement>() ?: return INVALID_INDEX
+        val descr = RustArgumentsDescription.findDescription(callArgs) ?: return INVALID_INDEX
+        var index = -1
+        if (descr.arguments.isNotEmpty()) {
+            index += generateSequence(callArgs.firstChild, { c -> c.nextSibling})
+                .filter { it.text == "," }
+                .takeWhile { it.textRange.startOffset < place.textRange.startOffset }
+                .count() + 1
+            if (index >= descr.arguments.size) {
+                index = -1
+            }
+        }
+        return index
+    }
+
+    private companion object {
+        val INVALID_INDEX: Int = -2
+    }
+}
+
+/**
+ * Holds information about arguments from func/method declaration
+ */
+class RustArgumentsDescription(
+    val arguments: Array<RustArgumentDescription>
+) {
+    fun getArgumentRange(index: Int): TextRange {
+        if (index < 0 || index >= arguments.size) return TextRange.EMPTY_RANGE
+        val start = (0..index - 1).sumBy { arguments[it].textLen + 2 }
+        val range = TextRange(start, start + arguments[index].textLen)
+        return range
+    }
+
+    val presentText = if (arguments.isEmpty()) "<no arguments>" else arguments.joinToString(", ")
+
+    companion object {
+        /**
+         * Finds declaration of the func/method and creates description of its arguments
+         */
+        fun findDescription(args: RustArgListElement): RustArgumentsDescription? {
+            val call = args.parent
+            val decl = when (call) {
+                is RustCallExprElement -> call.declaration
```

Comment:
If you write `call.declaration?.paramenters` then you wouldn't need the second `when`.